bindings/rust/src/lib.rs

```diff
@@ -68,29 +68,29 @@ type Fun = unsafe extern "C" fn(
 ) -> ::std::os::raw::c_int;
 
 pub fn zencode_exec(
-    script: String,
-    conf: String,
-    keys: String,
-    data: String,
+    script: &str,
```

Comment:
If you wanted to make the interface even easier you could do something like `script: impl AsRef<str>` and then you would do the same thing but wouldn't break backwards compatibility.

Reply:
I have tried your suggestion but I couldn't make the code compile. Since I have never used `AsRef` I might be doing something wrong, can you please try and share the patch/commit? Thank you!

Reply:
Try this snippet for the function `exec_f`, I think that it works!
```
fn exec_f(
    fun: Fun,
    script: impl AsRef<str>,
    conf: impl AsRef<str>,
    keys: impl AsRef<str>,
    data: impl AsRef<str>,
) -> Result<ZenResult, ZenError> {
    let mut stdout = Vec::<i8>::with_capacity(BUF_SIZE);
    let stdout_ptr = stdout.as_mut_ptr();
    let mut stderr = Vec::<i8>::with_capacity(BUF_SIZE);
    let stderr_ptr = stderr.as_mut_ptr();

    let lock = aquire_zen_gil();
    let exit_code = unsafe {
        fun(
            CString::new(script.as_ref())?.into_raw(),
            CString::new(conf.as_ref())?.into_raw(),
            CString::new(keys.as_ref())?.into_raw(),
            CString::new(data.as_ref())?.into_raw(),
            stdout_ptr,
            BUF_SIZE as u64,
            stderr_ptr,
            BUF_SIZE as u64,
        )
    };
    drop(lock);
```